src/main/java/org/broadinstitute/hellbender/utils/help/GATKWDLWorkUnitHandler.java

```diff
@@ -135,58 +143,142 @@ protected void processPositionalArguments(
         if (positionalArgsList != null && !positionalArgsList.isEmpty()) {
-            final Map<String, Object> positionalArgs = args.get("positional").get(0);
-            final String argType = (String) positionalArgs.get("type");
-            positionalArgs.put("testValue",
-                    getInputValueForTest(
-                            argType,
-                            (String) positionalArgs.get("defaultValue"))
+            final Map<String, Object> positionalArgs = args.get(TemplateProperties.ARGUMENTS_POSITIONAL).get(0);
+            positionalArgs.put(WDL_TEST_VALUE_PROPERTY,
+                    testValueAsJSON(
+                            WDLWorkUnitHandler.POSITIONAL_ARGS,
+                            clp.getPositionalArgumentDefinition(),
+                            (String) positionalArgs.get(TemplateProperties.ARGUMENT_TYPE),
+                            (String) positionalArgs.get(TemplateProperties.ARGUMENT_DEFAULT_VALUE))
             );
         }
     }
 
     /**
-     * Return a test input value for use in the WDL validation test inputs. If an option has WDL type
-     * File, then we need to provide the name of an actual file that exists so cromwell can localize it.
-     * "dummyWDLTestFileName" is a file that will be created by the test task.
+     * Return a test input value for use in the WDL validation test inputs.
+     *
+     * If an option has WDL type File, then we need to provide the name of an actual file that exists so cromwell
+     * can localize it: "dummyWDLTestFileName" is a file that will be created by the test task.
+     *
+     * @param longName the long name for this arg
+     * @param argDef the ArgumentDefinition for this arg
      * @param wdlType the wdl type for which an input value is needed
-     * @param defaultValue the default value for the argument for which an input value is required
+     * @param defaultWDLValue the default value for the argument for which an input value is required
      * @return a test input value that is either the default value, or the name of an actual test file
      * that will exist at test execution time
      */
-    protected String getInputValueForTest(final String wdlType, final String defaultValue) {
+    protected String testValueAsJSON(
+            final String longName,
+            final ArgumentDefinition argDef,
+            final String wdlType,
+            final String defaultWDLValue) {
+        final Argument argumentAnnotation = argDef.getUnderlyingField().getAnnotation(Argument.class);
+        final PositionalArguments positionalAnnotation = argDef.getUnderlyingField().getAnnotation(PositionalArguments.class);
+        final boolean isRequired =
+                (argumentAnnotation != null && !argumentAnnotation.optional())
+                || positionalAnnotation != null;
+        final String dummyWDLTestFile = ((GATKWDLDoclet) getDoclet()).getBuildDir() + "/" + dummyWDLTestFileName;
```

Comment:
Is it possible to use our `createTempFile()` method that automatically sets the file to be deleted on exit, and writes to the temp dir instead of the build dir?

Reply:
I update the build to clean up the outputs (including the cromwell output folders, as well as the dummy test file). Dynamically creating a temporary output directory would complicate things a bit because both (this) Java code and gradle validation code need to agree on the name. It could be done, but I've tested the cleanup I added and it seems to work. See what you think.  I expect the validation will mostly only be run on travis anyway.